internal/service/credentials/resource_credential_issuance_rule.go

```diff
@@ -0,0 +1,842 @@
+					stringvalidator.OneOf(
+						string(credentials.ENUMCREDENTIALISSUANCERULESTATUS_ACTIVE),
+						string(credentials.ENUMCREDENTIALISSUANCERULESTATUS_DISABLED)),
+				},
+			},
+
+			"filter": schema.SingleNestedAttribute{
+				Description:         filterDescription.Description,
+				MarkdownDescription: filterDescription.MarkdownDescription,
+				Optional:            true,
+				Attributes: map[string]schema.Attribute{
+					"group_ids": schema.SetAttribute{
+						ElementType: types.StringType,
+						Description: "Array of one or more identifiers (UUIDs) of groups, any of which a user must belong for the credential issuance rule to apply.",
+						Optional:    true,
+						Validators: []validator.Set{
+							setvalidator.ConflictsWith(path.MatchRelative().AtParent().AtName("population_ids")),
+							setvalidator.ConflictsWith(path.MatchRelative().AtParent().AtName("scim")),
+						},
+					},
+					"population_ids": schema.SetAttribute{
+						ElementType: types.StringType,
+						Description: "Array of one or more identifiers (UUIDs) of populations, any of which a user must belong for the credential issuance rule to apply. ",
+						Optional:    true,
+						Validators: []validator.Set{
+							setvalidator.ConflictsWith(path.MatchRelative().AtParent().AtName("group_ids")),
+							setvalidator.ConflictsWith(path.MatchRelative().AtParent().AtName("scim")),
+						},
+					},
+
+					"scim": schema.StringAttribute{
+						Description: "A SCIM query that selects users to which the credential issuance rule applies.",
+						Optional:    true,
+						Validators: []validator.String{
+							stringvalidator.ConflictsWith(path.MatchRelative().AtParent().AtName("group_ids")),
+							stringvalidator.ConflictsWith(path.MatchRelative().AtParent().AtName("population_ids")),
+						},
+					},
+				},
+			},
+
+			"automation": schema.SingleNestedAttribute{
+				Description: "Contains a list of actions, as key names, and the update method for each action.",
+				Required:    true,
+				Attributes: map[string]schema.Attribute{
+					"issue": schema.StringAttribute{
+						Description:         automationIssueDescription.Description,
+						MarkdownDescription: automationIssueDescription.MarkdownDescription,
+						Required:            true,
+					},
+					"revoke": schema.StringAttribute{
+						Description:         automationRevokeDescription.Description,
+						MarkdownDescription: automationRevokeDescription.MarkdownDescription,
+						Required:            true,
+					},
+					"update": schema.StringAttribute{
+						Description:         automationUpdateDescription.Description,
+						MarkdownDescription: automationUpdateDescription.MarkdownDescription,
+						Required:            true,
+					},
+				},
+			},
+
+			"notification": schema.SingleNestedAttribute{
+				Description: "Contains notification information. When this property is supplied, the information within is used to create a custom notification.",
+				Optional:    true,
```

Comment:
See comment below for optionals nested in optionals

Reply:
Resolved.

```
Validators: []validator.Object{
	objectvalidator.Any(
		objectvalidator.AlsoRequires(path.MatchRelative().AtName("methods")),
		objectvalidator.AlsoRequires(path.MatchRelative().AtName("template")),
	),
},
```